File: isuite-core-ui/src/main/java/gov/nwcg/isuite/core/service/MessageService.java
package gov.nwcg.isuite.core.service;

import gov.nwcg.isuite.core.vo.MessageBoardVo;
import gov.nwcg.isuite.core.vo.dialogue.DialogueVo;
import gov.nwcg.isuite.framework.core.service.TransactionService;
import gov.nwcg.isuite.framework.exceptions.ServiceException;

public interface MessageService extends TransactionService {
	
	public DialogueVo save(MessageBoardVo vo, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo getStaticMessage(DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo getPopUpGrid(DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo savePopUp(MessageBoardVo vo, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo getPopUpMessages(DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo deletePopUpMessage(MessageBoardVo vo, DialogueVo dialogueVo) throws ServiceException;

}

File: eis2/src/app/service/message.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs/Observable';
import { DialogueVo } from '../data/dialogue/dialoguevo';
import { MessageBoardVo } from 'src/app/data/message-board-vo';

@Injectable({
  providedIn: 'root'
})
export class MessageService {

  constructor(private http: HttpClient) { }

  getStaticMessage(): Observable<DialogueVo> {
    return this.http.get<DialogueVo>(`/service/messages/static`);
  }

  save(messageBoardVo: MessageBoardVo): Observable<DialogueVo> {
    return this.http.post('/service/messages', messageBoardVo);
  }

}
